src/main/java/org/broadinstitute/hellbender/engine/VariantWalker.java

```diff
@@ -57,7 +58,18 @@ void initializeFeatures() {
     public final SAMSequenceDictionary getBestAvailableSequenceDictionary() {
         final SAMSequenceDictionary dictFromDrivingVariants = drivingVariants.getSequenceDictionary();
         if (dictFromDrivingVariants != null){
-            return dictFromDrivingVariants;
+            //If this dictionary looks like it was synthesized from a feature index, see if there is
+            //a better dictionary available from another source (i.e., the reference)
+            final boolean isSyntheticDictionary = dictFromDrivingVariants
+                    .getSequences()
+                    .stream()
+                    .allMatch(seqRec -> seqRec.getSequenceLength() == SAMSequenceRecord.UNKNOWN_SEQUENCE_LENGTH);
```

Comment:
Maybe this should be extracted to a utility in the same class as the code that builds dictionaries from indices?